networkit/distance.pyx

```diff
@@ -924,6 +924,89 @@ cdef class APSP(Algorithm):
 		"""
 		return (<_APSP*>(self._this)).getDistance(u, v)
 
+cdef extern from "<networkit/distance/SPSP.hpp>":
+
+	cdef cppclass _SPSP "NetworKit::SPSP"(_Algorithm):
+		_SPSP(_Graph G, vector[node].iterator sourcesFirst, vector[node].iterator sourcesLast) except +
+		vector[vector[edgeweight]] getDistances() except +
+		edgeweight getDistance(node u, node v) except +
+		void setSources(vector[node].iterator sourcesFirst, vector[node].iterator sourcesLast)
+
+cdef class SPSP(Algorithm):
+	""" Some-Pairs Shortest-Paths algorithm (implemented running Dijkstra's algorithm from each source
+		node, or BFS if G is unweighted).
+
+    SPSP(G)
+
+    Computes pairwise shortest-path distances from the source nodes to all the nodes in G.
+
+    Parameters:
+	-----------
+	G : networkit.Graph
+		The graph.
+	sources : list.
+		Set of source nodes.
+    """
+	cdef Graph _G
+
+	def __cinit__(self, Graph G, sources):
```

Comment:
This should be `Graph G not None` to avoid that `None` is accepted as graph. Further, I am wondering why the second parameter is not simply `vector[node] sources`, in which case Cython would automatically cast the input and raise a `TypeError` if the wrong type is passed.

Reply:
Good idea, this way Cython takes care checking the type of `sources`. I updated the PR accordingly.